### PR TITLE
Cleanup and tests and post-as-bot

### PR DESCRIPTION
**Pull request by h3h**
Start organizing the code base more, and add unit tests.

Also switch to posting as the Slacker bot, rather than via slackbot’s incoming webhook integration.

### CODE REVIEW

**Comment by zeckalpha** [comment]
_gasp_ tests?

File: announcer.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def announce(self):
                 if self.slacker.channel_exists(config.announce_channel):
                     self.sb.say(config.announce_channel, m)
                 else:
-                    self.ds.warning("Attempted to announce in {}, but channel does not exist.".format(config.announce_channel))
+                    self.ds.logger.warning("Attempted to announce in %s, but channel does not exist.", config.announce_channel)
```
Comment (dlovitch) [inline announcer.py]: @h3h the % method of formatting is deprecated (https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations). Is there any reason not to stick with the format style?

Comment (h3h) [inline announcer.py]: Ah, interesting. I was following [the `logging` docs](https://docs.python.org/2/library/logging.html), which seem to have retained the `%s` format strings when values are being passed directly to a `Logger`.

That said, we could standardize on passing a fully `.format()`-ted string to all logger calls. I’m not sure if there are benefits or drawbacks to that vs. passing the values-to-be-formatted directly to the logger.

**Comment by h3h** [comment]
@dlovitch I also note that you linked to 3.x docs and I’ve been using 2.x. Do we have a target Python version? Should we have a `.python-version` file (or [`runtime.txt`](https://devcenter.heroku.com/articles/python-runtimes)?) designating such?

**Comment by dshackith** [comment]
I can tell you that this does not run out of the box with Python 2.6, but does run with 2.7 that is installed on OS X 10.12

File: utils/__init__.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import codecs
+
+def get_local_file_content(file_name):
+    """Read the contents of `file_name` into a unicode string, return the unicode string."""
+    f = codecs.open(file_name, encoding='utf-8')
+    ret = f.read().strip()
+    f.close()
```
Comment (zeckalpha) [inline utils/__init__.py]: Does `with` work with `codecs.open`?

**Comment by zeckalpha** [comment]
I've not used it in a while, but is `tox` still the thing to use to multiple python versions?

File: announcer.py (hunk 1 above)
Comment (h3h) [inline announcer.py]: @dlovitch Do you have an opinion on how we move forward on this string/log formatting question? I don’t have a strong opinion either way.

File: utils/__init__.py (hunk 2 above)
Comment (h3h) [inline utils/__init__.py]: I think so. Though now that I look at it, this should likely be using a `try`/`except` to catch missing/unreadable files.

**Comment by h3h** [comment]
@zeckalpha Added 4 more commits. Re-check?

**Comment by h3h** [comment]
Notable change in 0a0db069856930261ce5092b8beff9ac270dfc45 to start posting as the bot configured for `Slacker` rather than via slackbot. Rationale included within.

File: announcer.py (hunk 1 above)
Comment (zeckalpha) [inline announcer.py]: My only thought would be whatever works in both 2.7 and 3+

**Comment by zeckalpha** [comment]
Worst case, we revert!

**Pull request merged**